src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeImageSelect.ts

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { getImageUrl } from 'src/analysis/modal-utils';
+import { ComputeImage, ComputeImageStore, useComputeImages } from 'src/analysis/useComputeImages';
+import { RuntimeToolLabel, runtimeTools } from 'src/analysis/utils/tool-utils';
+import { spinnerOverlay } from 'src/components/common';
+import { GetRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+
+export const defaultImageId = 'terra-jupyter-gatk';
+
+export interface GcpComputeImageSelectProps {
+  readonly setSelectedImage: (image: ComputeImage | undefined) => void;
+  readonly setIsCustomSelectedImage: (value: boolean) => void;
+  readonly tool: RuntimeToolLabel;
+  readonly currentRuntime?: GetRuntimeItem;
+}
+
+const customImageUrl = 'CUSTOM_IMAGE_URL';
+
+export const GcpComputeImageSelect: React.FC<GcpComputeImageSelectProps> = (props: GcpComputeImageSelectProps) => {
+  const { loadedState, refreshStore }: ComputeImageStore = useComputeImages();
+  const [images, setImages] = useState<ComputeImage[]>([]);
+  const [selectedComputeImageUrl, setSelectedComputeImageUrl] = useState<string>('');
+
+  _.once(refreshStore());
+
+  const getImageOptions = (predicate) =>
+    _.flow(
+      _.filter(predicate),
+      _.map(({ label, image }) => ({ label, value: image }))
+    )(images);
+
+  const renderImageSelect = () => {
```

Comment:
These types of renderX methods are discouraged since they bloat the performance-sensitive render code here.  I always ask to push a renderX method like this out to a separate React.FC that is declared before the main React.FC here using it.  Not only will it yield better performance, it will lighten up the code in the main FC.